ci/test.sh

```diff
@@ -60,7 +60,32 @@ run_test() {
 		RUNNER="$TEST_CMD"
 	fi
 
-	eval $RUNNER || failure
+	if [[ "$GITTEST_FLAKY_RETRY" > 0 ]]; then
+		ATTEMPTS_REMAIN=$GITTEST_FLAKY_RETRY
+	else
+		ATTEMPTS_REMAIN=1
+	fi
+
+	while [[ "$ATTEMPTS_REMAIN" > 0 ]]; do
+		if [ "$FAILED" -eq 1 ]; then
+			echo ""
+			echo "Re-running flaky ${1} tests..."
+			echo ""
+		fi
+
+		FAILED=0
+		eval $RUNNER || FAILED=1 && true
+
+		if [ "$FAILED" -eq 0 ]; then
+			break
+		fi
+
+		ATTEMPTS_REMAIN="$(($ATTEMPTS_REMAIN-1))"
+	done
+
+	if [ "$FAILED" -ne 0 ]; then
+		SUCCESS=0
+	fi
```

Comment:
Previously we were executing `failure`, don't we have to do that now, too?

Reply:
Ach, that ended up dead.  I removed it, but I also improved the error message a little, inspired by that function.